mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp

```diff
@@ -617,6 +632,40 @@ bool CNEOBaseCombatWeapon::CanBePickedUpByClass(int classId)
 	return true;
 }
 
+#ifdef GAME_DLL
```

Comment:
Seems like there's no need for the server to calculate the flashes & send them over the wire here - couldn't the clients themselves do this work for the events they are locally aware of?

Reply:
I was following this tutorial https://developer.valvesoftware.com/wiki/Particle_Muzzle_Flash which is why its done the way it is, but I can probably see what function is fired on the clientside when the particleeffect is dispatched from the server, and call it in this function itself only on the client side

Reply:
Figured it out